src/richie/apps/search/index_manager.py

```diff
@@ -127,8 +128,13 @@ def perform_aliases_update():
             # attempting to create. In Richie, this is not supposed to happen and is usually the
             # result of a broken ES state.
             if exception.error == "invalid_alias_name_exception":
-                # Identify the broken index
-                broken_index = exception.info["error"]["index"]
+                try:
+                    # ES6 provides the name of the broken index directly
+                    broken_index = exception.info["error"]["index"]
```

Comment:
I'm wondering if we could instead get the index name from the alias?

Reply:
This information is unavailable in the exception. Here's what it looks like in ES7:
```
{'error': {'root_cause': [{'type': 'invalid_alias_name_exception', 'reason': 'Invalid alias name [richie_test_courses]: an index or data stream exists with the same name as the alias'}], 'type': 'invalid_alias_name_exception', 'reason': 'Invalid alias name [richie_test_courses]: an index or data stream exists with the same name as the alias'}, 'status': 400}
```

Also we're performing a bunch of updates all at once so the list of actions does not tell us which index is broken either.

I don't see another way than using a regex on the error here.

Reply:
ok then. Let's keep it like this and see how it works.